core/domain/opportunity_domain_test.py

```diff
@@ -0,0 +1,349 @@
+            self._assert_validation_error(
+                self.valid_exp_opprtunity_summary,
+                'Invalid language_code: invalid_language_code')
+
+    def test_translation_counts_with_invalid_count_fails_validation(
+            self):
+        self.valid_exp_opprtunity_summary.translation_counts = {
+            'hi': 4
+        }
+        with self.mock_supported_audio_languages_context:
+            # Object with valid language_code in translation_counts passes the
+            # validation.
+            self.valid_exp_opprtunity_summary.validate()
+            self.valid_exp_opprtunity_summary.translation_counts = {
+                'hi': -5
+            }
+            # Object with invalid language_code in translation_counts fails the
+            # validation.
+            self._assert_validation_error(
+                self.valid_exp_opprtunity_summary,
+                'Expected count for language_code hi to be an positive integer,'
+                ' received -5')
+
+    def test_translation_counts_with_invalid_count_type_fails_validation(
+            self):
+        self.valid_exp_opprtunity_summary.translation_counts = {
+            'hi': 4
+        }
+        with self.mock_supported_audio_languages_context:
+            # Object with valid count in translation_counts passes the
+            # validation.
+            self.valid_exp_opprtunity_summary.validate()
+            self.valid_exp_opprtunity_summary.translation_counts = {
+                'hi': '12ab'
+            }
+            # Object with invalid count in translation_counts fails the
+            # validation.
+            self._assert_validation_error(
+                self.valid_exp_opprtunity_summary,
+                'Expected count for language_code hi to be an integer, '
+                'received 12ab')
+
+    def test_translation_counts_with_invalid_count_value_fails_validation(
+            self):
+        self.valid_exp_opprtunity_summary.content_count = 5
+        self.valid_exp_opprtunity_summary.translation_counts = {
+            'hi': 4
+        }
+        with self.mock_supported_audio_languages_context:
+            # Object with valid count value i.e, less than or equal to
+            # content_count in translation_counts passes the validation.
+            self.valid_exp_opprtunity_summary.validate()
+            self.valid_exp_opprtunity_summary.translation_counts = {
+                'hi': 8
+            }
+            # Object with invalid count value i.e, more than content_count
+            # in translation_counts fails the validation.
+            self._assert_validation_error(
+                self.valid_exp_opprtunity_summary,
+                'Expected translation count for language_code hi to be '
+                r'less than or equal to content_count\(5\), received 8')
+
+    def test_incomplete_translation_languages_with_invalid_language_code_fails_validation( # pylint: disable=line-too-long
```

Comment:
test_invalid_lang_code_in_incomplete_translation_langs_fails_validation(

(not pretty, but I think it works)

Reply:
You're actually testing validation, and not so much the act of "changing" things, so I think what I have works fine (and fits in the line limit, which was what I was trying to do).